sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/administration/models/RuleProperties.java

```diff
@@ -0,0 +1,270 @@
+                if (model instanceof TrueRuleFilter) {
+                    return trueFilter;
+                } else if (model instanceof FalseRuleFilter) {
+                    return falseFilter;
+                } else if (model instanceof CorrelationRuleFilter) {
+                    final CorrelationRuleFilter filter = (CorrelationRuleFilter) model;
+                    final CorrelationFilterImpl returned = new CorrelationFilterImpl()
+                        .setContentType(filter.getContentType())
+                        .setCorrelationId(filter.getCorrelationId())
+                        .setLabel(filter.getLabel())
+                        .setMessageId(filter.getMessageId())
+                        .setTo(filter.getTo())
+                        .setSessionId(filter.getSessionId())
+                        .setReplyTo(filter.getReplyTo())
+                        .setReplyToSessionId(filter.getReplyToSessionId());
+
+                    if (!filter.getProperties().isEmpty()) {
+                        final List<KeyValueImpl> parameters = filter.getProperties().entrySet()
+                            .stream()
+                            .map(entry -> new KeyValueImpl()
+                                .setKey(entry.getKey()).setValue(entry.getValue().toString()))
+                            .collect(Collectors.toList());
+
+                        returned.setProperties(parameters);
+                    }
+
+                    return returned;
+                } else if (model instanceof SqlRuleFilter) {
+                    final SqlRuleFilter filter = (SqlRuleFilter) model;
+                    final SqlFilterImpl returned = new SqlFilterImpl()
+                        .setSqlExpression(filter.getSqlExpression())
+                        .setCompatibilityLevel(filter.getCompatibilityLevel())
+                        .setRequiresPreprocessing(filter.getRequiresPreprocessing());
+
+                    if (!filter.getProperties().isEmpty()) {
+                        final List<KeyValueImpl> parameters = filter.getProperties().entrySet()
+                            .stream()
+                            .map(entry -> new KeyValueImpl()
+                                .setKey(entry.getKey()).setValue(entry.getValue().toString()))
+                            .collect(Collectors.toList());
+
+                        returned.setParameters(parameters);
+                    }
+
+                    return returned;
+                } else {
+                    return null;
+                }
+            }
+        });
+    }
+
+    /**
+     * Initializes a new instance with the given rule {@code name}, {@code filter}, and {@code action}.
+     *
+     * @param name Name of the rule.
+     * @param filter Filter for the rule.
+     * @param action Action for the rule.
+     */
+    RuleProperties(String name, RuleFilter filter, RuleAction action) {
+        this.name = name;
```

Comment:
Should `filter` be checked for not null in constructor ? 
Also is null `action` allowed here ?

Reply:
This is package private because the service returns the values.